lib/cache/access_list.go

```diff
@@ -39,11 +39,15 @@ func newAccessListCollection(upstream services.AccessLists, w types.WatchKind) (
 	}
 
 	return &collection[*accesslist.AccessList, accessListIndex]{
-		store: newStore(map[accessListIndex]func(*accesslist.AccessList) string{
-			accessListNameIndex: func(al *accesslist.AccessList) string {
-				return al.GetMetadata().Name
+		store: newStore(
+			func(al *accesslist.AccessList) *accesslist.AccessList {
+				return al.Clone()
 			},
```

Comment:
Why not just the method, here and in the other few places where we use a func expression? It's essentially the same (functions with a receiver are still boxed IIRC), it's just shorter.

```suggestion
			(*accesslist.AccessList).Clone,
```